src/rust/catmem/queue.rs

```diff
@@ -113,12 +113,19 @@ impl CatmemQueue {
         {
             let mut ring: RefMut<Ring> = self.ring.borrow_mut();
             ring.prepare_close()?;
-            ring.close()?;
-            ring.commit();
-            ring.prepare_closed()?;
-            ring.commit();
```

Comment:
We should abort if we fail on `self.close()`, otherwise the state machine may end up have an invalid `next` state.